03week/towersOfHanoi.js

```diff
@@ -13,31 +18,67 @@ let stacks = {
   c: []
 };
 
+const piece = [4, 3, 2, 1];
+const stackArrA = stacks.a;
+const stackArrB = stacks.b;
+const stackArrC = stacks.c;
+let stacksArr = Object.values(stacks);// returns stacks property values
+let stacksProps = Object.keys(stacks);// returns keys as array ['a','b','c']
+let stacksArrLength = stacksArr.length;
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
-
-function movePiece() {
-  // Your code here
+// Only remove one piece at a time from top of any startStack
+// Remove last element only
+function movePiece(piece, startStack, endStack){
+  // endStack.push(startStack.pop());
+  // console.log('startStack => endStack');
 
 }
-
-function isLegal() {
-  // Your code here
-
+// Accepts two inputs (towers) one from startStack and one from endStack
+// Will return false if piece from startStack is larger than the piece on the endStack
+// Only movePiece to either an empty 'stacks' or to a 'stacks' with a larger piece already present  
+function isLegal(startStack, endStack) {
+  if(startStack.length == 0){
+    return false;
+  }else if(endStack.length == 0){
+    return true;
+  } 
+  let L1 = startStack.length-1;
+  let L2 = endStack.length-1;
+    if(startStack[L1] < startStack[L2]){
+      return false;
+    } else {
+      return true;
+    }
 }
 
+// All disks must be appropriately stacked at 'stacks.c' for winning position
 function checkForWin() {
-  // Your code here
-
+  if(stacksArrC == winner){
+    return `You have completed the Towers of Hanoi!!!`;
+    console.log('checkedForWin');
+  } 
 }
 
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+  const gameStart = [0, 0, 2, 0, 1, 1];
+  const endStart = [2, 1, 1, 2, 0, 2];
+  //for each element in start array, 
+  for(let i = 0; i < startStack.length; i++){
```

Comment:
use a forEach, no need to increase the scope with a for loop

Reply:
also, why do you need a loop here, you have the start and the destination, you wouldn't want to loop through each, the user would dictate